### PR TITLE
Voila 0.5 migration

### LINKED ISSUE #91: RFE: please provide update for `voila` 0.5.3
https://github.com/voila-dashboards/voici/blob/1587e7bfa2a87b71be568a20e385dd9d8da6e2ae/pyproject.toml#L33-L34

### PR DESCRIPTION
**Pull request by trungleduc**
## References

Fix #91

https://github.com/voila-dashboards/voici/assets/4451292/0633d991-2a19-4019-a9e5-034efabbcdb2

- Update frontend to JupyterLab 4, Voila 0.5.0 components
- Tree page is now a Lab remix app.
- New argument `--classic-tree` to use the jinja2-based tree page instead of the Lab-based one.
- Voila CLI style CLI now supports the [`files` directory](https://jupyterlite.readthedocs.io/en/latest/howto/content/files.html#content-with-the-cli) to copy additional files to the Voici deployment
  -  `voici .`
  - `voici notebook.ipynb`

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -31,7 +31,7 @@ classifiers = [
     "Programming Language :: Python :: 3.11",
 ]
 dependencies = [
-    "voila==0.5.0a4",
+    "voila==0.5.0",
```
Comment (martinRenou) [inline pyproject.toml:34]:
```suggestion
    "voila>=0.5,<0.6",
```

**Review by martinRenou — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -35,6 +35,13 @@
       "pre-commit": "lint-staged"
     }
   },
+  "resolutions": {
+    "@jupyterlite/contents": "file:/home/trungle/WORK/jupyterlite/packages/contents",
```
Comment (jtpio) [inline package.json:39]: FYI it should now be possible to build against released packages: https://www.npmjs.com/package/@jupyterlite/contents/v/0.2.0-alpha.0

https://github.com/jupyterlite/jupyterlite/releases/tag/v0.2.0a0

**Comment by martinRenou** [comment]
Fixes #91

File: packages/voici/src/main.ts [hunk 3]
```diff
@@ -166,10 +169,10 @@ async function main() {
   const kernelspecs = await jupyterLiteServer.resolveRequiredService(
     IKernelSpecs
   );
-
   const serviceManager = jupyterLiteServer.serviceManager;
+
   const app = new VoiciApp({
-    serviceManager: serviceManager as any,
+    serviceManager: serviceManager,
```
Comment (jtpio) [inline packages/voici/src/main.ts:175]: Maybe something more succinct:

```suggestion
    serviceManager,
```

Comment (trungleduc) [reply packages/voici/src/main.ts:175]: Thanks!

**Comment by trungleduc** [comment]
update playwright snapshots

File: packages/voici/src/plugins/widgets.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+ * The Voici widgets manager plugin.
+ */
+export const widgetManager = {
+  id: '@voila-dashboards/voici:widget-manager',
+  autoStart: true,
+  provides: base.IJupyterWidgetRegistry,
+  activate: async (app: JupyterFrontEnd): Promise<any> => {
+    if (!(app instanceof VoiciApp)) {
+      throw Error(
+        'The Voici Widget Manager plugin must be activated in a VoilaApp'
```
Comment (martinRenou) [inline packages/voici/src/plugins/widgets.ts:24]:
```suggestion
        'The Voici Widget Manager plugin must be activated in a VoiciApp'
```

File: tests/test_app.py
Comment (martinRenou) [inline tests/test_app.py:1]: 🚀

File: voici/addon.py [hunk 5]
```diff
@@ -25,15 +25,6 @@
 
 from .tree_exporter import VoiciTreeExporter
 
-# Try to patch xeus-python's jupyter-widgets version st. we pull the right one
-try:
-    from jupyterlite_xeus_python.env_build_addon import XeusPythonEnv
-
-    if hasattr(XeusPythonEnv, "pin_packages"):
-        XeusPythonEnv.pin_packages = ["ipywidgets=8.0.6", "jupyterlab_widgets=3.0.7"]
```
Comment (martinRenou) [inline voici/addon.py:33]: 🚀

File: voici/app.py [hunk 6]
```diff
@@ -25,6 +24,7 @@
     strip_sources="VoilaConfiguration.strip_sources",
     template="VoilaConfiguration.template",
     theme="VoilaConfiguration.theme",
+    **{"classic-tree": "VoilaConfiguration.classic_tree"},
```
Comment (martinRenou) [inline voici/app.py:27]:
```suggestion
    "classic-tree"="VoilaConfiguration.classic_tree",
```

File: voici/tree_exporter.py [hunk 7]
```diff
@@ -71,15 +95,21 @@ def __init__(
         voici_configuration: VoilaConfiguration,
         **kwargs,
     ):
-        self.jinja2_env = jinja2_env
+        self.jinja2_env = self._environment_cached = jinja2_env
```
Comment (martinRenou) [inline voici/tree_exporter.py:98]:
```suggestion
        self.jinja2_env = jinja2_env
```

?

**Review by martinRenou — CHANGES_REQUESTED** [review]
A couple of minor comments, otherwise it looks great! Thanks!

File: voici/app.py (hunk 6 above)
Comment (trungleduc) [reply voici/app.py:27]: I need to use this trick because of the hyphen in the parameter name

File: voici/tree_exporter.py (hunk 7 above)
Comment (trungleduc) [reply voici/tree_exporter.py:98]: Needs the `self._environment_cached` in order to make `self._init_resources()` works

File: .github/workflows/main.yml [hunk 8]
```diff
@@ -39,18 +39,18 @@ jobs:
...
       working-directory: ui-tests
 
-    - name: Build UI tests
-      run: yarn run build
-      working-directory: ui-tests
+    - name: Install dependencies and build tests
+      run: |
+        python -m pip install .
+        cd ui-tests
+        yarn install --frozen-lockfile
```
[the remark this answers is not in the document]
Comment (trungleduc) [reply .github/workflows/main.yml]: Thanks!

**Review by martinRenou — APPROVED** [review]
Thank you!

**Pull request merged**